### DISCUSSION TITLE
Attaching a file to an email before sending it?

### DISCUSSION THREAD

**Discussion by mrabb24** [sadieblue/agilityblue.workspace-scripts#1] [Q&A] [answered]
Is it possible to have a script automatically include an attachment from the project onto the email template being used to send the email?

Thoughts?

**Comment by bmward80** [comment]
Hi Matt, this should be possible using the [Add-EmailMessageAttachment](https://help.agilityblue.com/docs/scripting-add-emailmessageattachment) cmdlet. I'll see if I can come up with an example of that for you.

**Comment by bmward80** [comment, marked as answer]
Hi Matt,

Here's an example that works when you use the before save state on a send email trigger. The file id doesn't have to be specifically from a project - it can be any attachment within the workspace.

```powershell
$email = Get-InboundObjectInstance

$listOfAttachments = @(
    @{ 
        EmailMessageId = $email.EmailMessageId
        FileId = "5538f7e9b2e748e2b6b223d56155ca20"
    }
)

Add-EmailMessageAttachments -Entries $listOfAttachments | Out-Null

return $email
```

Hope this helps!
Brandon

**Reply by mrabb24** [reply]
perfect, thanks!
